### ISSUE TITLE
KeyError at /accounts/weibo/login/callback/

### ISSUE THREAD

**Issue by Tanych** [pennersr/django-allauth#1375]
It works well when signing up with the Weibo app management username and password. However, when I try to sign up using others Weibo accounts and get the authorization from the apps. It will raise KeyError due to the absence of key "idstr" in data in WeiboProvider.extract_uid().

``` python
Request Method:     GET
Request URL:    http://127.0.0.1:8080/accounts/weibo/login/callback/?state=vZO8wmTjF3DP&code=c6a9d9ba4a4b3bdfd4e3393a92f5f039
Django Version:     1.8.8
Exception Type:     KeyError
Exception Value:    

'idstr'

Exception Location:     /usr/local/lib/python2.7/dist-packages/allauth/socialaccount/providers/weibo/provider.py in extract_uid, line 25
Python Executable:  /usr/bin/python
Python Version:     2.7.6
```

**Comment by nujabes8** [comment]
Following view class only catches PermissionDenied, OAuth2Error
* OAuthCallbackView
* OAuth2CallbackView
However we get following error on production level when there are swarming request or undistinguishable api return from providers. 

```
HTTPError at /accounts/facebook/login/callback/

500 Server Error: Internal Server Error
```

```
HTTPError at /accounts/google/login/callback/

500 Server Error: Internal Server Error
```

```
ConnectionError at /accounts/twitter/login/callback/

HTTPSConnectionPool(host='api.twitter.com', port=443): Max retries exceeded with url: /1.1/account/verify_credentials.json?include_email=true (Caused by <class 'socket.error'>: [Errno 110] Connection timed out)

```

```
TypeError at /accounts/facebook/login/callback/

'NoneType' object has no attribute '__getitem__'
````

**Comment by pennersr** [comment]
The Weibo case is closed via 2082160e915371f1ff03f91c3bb55855811398b3

Not sure what needs to be done for VK, please file a pull request for that using the approach taken for Weibo.

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/1375
